### ISSUE TITLE
NullReferenceException in DiagnosticListener constructor

### ISSUE THREAD

**Issue by lmolkova** [dotnet/corefx#20306]
1.	If someone listens to DiagnosticSourceEventSource bridge
2.	When any DiagnosticListener is created, it causes static creation of DiagnosticSourceEventSource
3.	When it’s created, OnEventCommand is fired that eventually causes call to DiagnosticListener.AllListeners
4.	Which creates HttpHandlerDiagnosticListener -> DiagnosticListener that fails on DiagnosticSourceEventSource.Logger.IsEnabled (we are still in the process of creation DiagnosticSourceEventSource.Logger and it’s still null)

```
Unhandled Exception: System.TypeInitializationException: The type initializer for 'System.Diagnostics.HttpHandlerDiagnosticListener' threw an exception. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Diagnostics.DiagnosticListener..ctor(String name)
   at System.Diagnostics.HttpHandlerDiagnosticListener..ctor()
   at System.Diagnostics.HttpHandlerDiagnosticListener..cctor()
   --- End of inner exception stack trace ---
   at System.Diagnostics.DiagnosticListener.get_AllListeners()
   at ConsoleApp3.Program.Main(String[] args) in C:\repo\delme\ConsoleApp3\Program.cs:line 13
```

Code to reproduce
```c#
   class Program : IObserver<DiagnosticListener>
    {
        static void Main(string[] args)
        {
            new MyEventListener();
            new DiagnosticListener("123");
            DiagnosticListener.AllListeners.Subscribe(new Program());
        }

        public void OnNext(DiagnosticListener value) {}
        public void OnError(Exception error){}
        public void OnCompleted() {}

        class MyEventListener : EventListener
        {
            protected override void OnEventSourceCreated(EventSource eventSource)
            {
                if (eventSource != null && eventSource.Name == "Microsoft-Diagnostics-DiagnosticSource")
                {
                    this.EnableEvents(eventSource, EventLevel.Verbose, EventKeywords.All);
                }

                base.OnEventSourceCreated(eventSource);
            }

            protected override void OnEventWritten(EventWrittenEventArgs eventData)
            {
            }
        }
```

**Comment by karelz** [comment]
@lmolkova I assume this is important for your team - do you have plan to address it in rel/2.0.0 by Friday? We are in Escrow afterwards and bug fixes will get even more scrutiny ...

**Comment by stephentoub** [comment]
Fix in #20316 was merged; keeping this open to track porting it to release/2.0.0.

**Comment by karelz** [comment]
@vancem are you working on a port to release/2.0.0 branch? You have to then send email to netcore shiproom for approval. Let me know if you need help with the process.

**Comment by stephentoub** [comment]
Fixed by #20337
